unstructured/ingest/interfaces.py

```diff
@@ -61,6 +73,53 @@ class ProcessorConfig(BaseConfig):
     num_processes: int = 2
 
 
+@dataclass
+class FsspecConfig(BaseConfig):
```

Comment:
what's the thinking around moving this to interfaces?

Reply:
got it, I guess the icky-ness is that this is for a specific implementation: fsspec. though...really does this have to be fsspec? seems like these options are all reasonable for any file storage type connector? e.g. onedrive and google drive which are both file storage sources but not fsspec

Reply:
...not that we need to support these in this PR, but that the name of this config could be more abstract if it could be used to satisfy those as well in a later PR

Reply:
Small modification made to make it more generic. Some of the code is still specific to fsspec such as all the protocol parsing but I moved out what could be generic into a FileStorageConfig.